src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -271,6 +275,33 @@ message GetClusterStatusReply {
   ClusterResourceState cluster_resource_state = 2;
 }
 
+enum DrainNodeReason {
+  // The node has been idle for long enough time.
+  // Drain request with this reason is rejectable.
+  IDLE_TERMINATION = 0;
+  // The node will be preempted.
+  // Drain request with this reason is not rejectable.
+  PREEMPTION = 1;
+}
+
+message DrainNodeRequest {
+  // The id of the node to be drained.
+  bytes node_id = 1;
+  // The reason why the node will be drained.
+  // This information will be used to
+  // decide whether the request is rejectable.
+  DrainNodeReason reason = 2;
+  // The detailed drain reason message.
+  // Used for observability.
+  string reason_message = 3;
+}
+
+message DrainNodeReply {
```

Comment:
this should be part of gcs state to autoscaler. It should include instance_id.

Reply:
i guess we decided to not have drain request as part of autoscaler's sync message? cc @scv119

Reply:
i think for completeness we should have the instance_id in the response.

Reply:
The reply is good. And in the cluster report, the node status also changes to DRAINED. PA will only check the cluster report.